### DISCUSSION TITLE
Typescript docs for module extension not working?

### DISCUSSION THREAD

**Discussion by vymao** [nextauthjs/next-auth#6166] [Help]
### Question 💬

I am trying to follow the [Typescript docs](https://next-auth.js.org/getting-started/typescript), but in VSCode, I still get an error regarding properties not being recognized. 

### How to reproduce ☕️

I am trying to follow the Next-Auth [docs][1] that describe how to extend the session interface in Typescript. Specifically, I want to add the accessToken property to pass to the front-end. 

I have created a `next.d.ts` file under `types/` with the following: 

```
import NextAuth, { Account, DefaultSession, User } from "next-auth"
import { JWT } from "next-auth/jwt"

declare module "next-auth" {
    interface Session {
        accessToken?: Account.accessToken
    }
}


declare module "next-auth/jwt" {
    interface JWT {
        accessToken?: Account.accessToken
    }
}
```
I have also added `typeRoots` to my tsconfig.json: `"typeRoots": ["./types/", "./node_modules/@types/"]`, as the docs suggest. However, when I try and read the accessToken, 
```
  const { data: session } = useSession()
  const { accessToken } = session
```
I still get the following error: 
```
Property 'accessToken' does not exist on type 'Session | null'.
```

  [1]: https://next-auth.js.org/getting-started/typescript

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
![image](https://user-images.githubusercontent.com/18369201/209414761-974aad7f-678c-4a09-95a4-96990d653cbe.png)
works fine for me. Hard to say anything more without a reproduction. Although one issue might be that you are trying to destructure a value that is possibly `null`, which won't work. You need to first check if data is there.
